### PR TITLE
Handling 100 req/s blog post

### PR DESCRIPTION
**Pull request by davidfischer**
- Talk about load testing
- Talk about comparing services
- Discuss some future challenges

The audience for this blog is general tech folks and hopefully that will result in more publishers longer term.

### CODE REVIEW

File: content/posts/2021-hundred-requests-per-second-with-django.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+Django and Python may be slow but at most replacing them could shave 10-15ms off our performance.
+Improvements on our infrastructure or how we handle data would yield better results.
+
+<div class="postimage">
+ <img class="w-100" src="{static}../images/posts/2021-adserver-performance-breakdown.png" alt="Performance breakdown of the ad server">
+ <p>A little over half our server time is spent in Postgres while Python is just under half</p>
+</div>
+
+
+## Current challenges
```
Comment (davidfischer) [inline content/posts/2021-hundred-requests-per-second-with-django.md:125]: This section is a bit long and doesn't really deal with performance and 100 req/s. Thoughts on cutting it down?

File: content/posts/2021-hundred-requests-per-second-with-django.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+to our publishers and advertisers.
+We've done a lot of the low hanging optimizations here but we will need to break our data storage
+into our transactional database and a system used for reporting, aggregation, and archival.
+Anytime we need to add more infrastructure, it's never an easy problem.
+
+Another big challenge we hope to solve in the longer term is how to have good performance on different continents.
+We can optimize server performance all we want but in Europe and Asia performance is about 100ms worse
+due purely to latency.
+Currently, our source of truth is what is in our Postgres database
+and reworking our solution to actually process ad requests at the edge will be a big change.
```
Comment (ericholscher) [inline content/posts/2021-hundred-requests-per-second-with-django.md:143]: Should we note that every request also does a write, which makes this much harder. Definitely an interesting blog post in the future, if we ever implement it.

File: content/posts/2021-hundred-requests-per-second-with-django.md (hunk 1 above)
Comment (ericholscher) [reply content/posts/2021-hundred-requests-per-second-with-django.md:125]: Yea, I'd cut it down a bit, and maybe put it at the end?

**Review by ericholscher — APPROVED** [review]
Looks great. I'd love to have a bit more of an orchestrated release of this post on HN, doing maybe a tweet and some promotion to get folks directed towards it?

File: content/posts/2021-hundred-requests-per-second-with-django.md (hunk 1 above)
Comment (davidfischer) [reply content/posts/2021-hundred-requests-per-second-with-django.md:125]: Well, it is at the end. I will try to cut it down though.

File: content/posts/2021-hundred-requests-per-second-with-django.md (hunk 2 above)
Comment (davidfischer) [reply content/posts/2021-hundred-requests-per-second-with-django.md:143]: This blog isn't a perfect fit but maybe I'll just mention multi-writer scenarios.

File: content/posts/2021-hundred-requests-per-second-with-django.md (hunk 1 above)
Comment (davidfischer) [reply content/posts/2021-hundred-requests-per-second-with-django.md]: I ended up removing the section about PG data accumulation and just telling the story about multi-writer scenarios. I think it's a bit more compelling anyway.

**Comment by davidfischer** [comment]
When are you thinking about rolling this out?

I can post to HN and maybe an accompanying tweet like:

```
We scaled our #Django powered ad server to 100 write requests per second using off-the-shelf tools while keeping our cloud costs down. Check it out!
```

**Pull request merged**